src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -68,7 +68,10 @@ internal partial class CSharpIntroduceVariableService
                     // this will be null for expression-bodied properties & indexer (not for individual getters & setters, those do have a symbol),
                     // both of which are a shorthand for the getter and always return a value
                     var method = document.SemanticModel.GetDeclaredSymbol(arrowExpression.Parent, cancellationToken) as IMethodSymbol;
-                    var createReturnStatement = !method?.ReturnsVoid ?? true;
+                    var createReturnStatement = true;
+
+                    if (method is not null)
+                        createReturnStatement = !method.ReturnsVoid && !method.IsAsyncReturningVoidTask(document.SemanticModel.Compilation);
```

Comment:
```suggestion
                    var createReturnStatement =
                        method != null &&
                        !method.ReturnsVoid &&
                        !method.IsAsyncReturningVoidTask(document.SemanticModel.Compilation);
```

Reply:
In this suggestion `createReturnStatement` won't be `true` when `method` is `null`, which is required here (read explanation comment above). This can be refactored to a single statement, but it would contain both `||` and `&&`, which hurts readability